internal/iam/query.go

```diff
@@ -206,6 +206,102 @@ const (
         on grants.role_id = roles.role_id;
     `
 
+	grantsForUserGlobalResourcesQuery = `
+	with
+    users (id) as (
+      select public_id
+        from iam_user
+      %s -- anonUser || authUser
```

Comment:
I believe we can eliminate the `%s` here by making this something like:

```sql
   where public_id = any(@user_ids)
```

And then when we use this query, constructing the args like:

```go
var args []any
switch userId {
case globals.AnonymousUserId:
  args = append(args, sql.Named("user_ids", "{'u_anon', 'u_auth'}"))
default:
  args = append(args, sql.Named("user_ids", fmt.Sprintf("{'%s', 'u_anon', 'u_auth'}", userId)))
}
```

Reply:
Updated to use `sql.Named()` for the userId & resource query args: 5e8d9fb8bdb83b940ea6297fbb1f4c3d27b86596